### PR TITLE
Correction of Mechanical Ventilation Controller Index Problem

### LINKED ISSUE #9558: Incorrect zone volume for Controller:MechanicalVentilation ACH
Issue overview
--------------
While reviewing code changes for #9553 noticed [this line in `MixedAir::CalcMechVentController`](https://github.com/NREL/EnergyPlus/blob/5f56911f1857b1d794f61d9ba4e085f3bd33c346/src/EnergyPlus/MixedAir.cc#L4122-L4123):
```
                    ZoneOAACH = curZone.Multiplier * curZone.ListMultiplier *
                                (this->ZoneOAACHRate(ZoneIndex) * state.dataHeatBal->Zone(ZoneIndex).Volume) * curZoneOASchValue / 3600.0;
```
`state.dataHeatBal->Zone(ZoneIndex).Volume` should be `curZone.Volume`
Controller:MechanicalVentilation has data for a list of zones, and `ZoneIndex` walks through that list. So, member variables of `VentilationMechanical` should use `ZoneIndex`, but anything that looks into `Zone` or other places outside of `VentilationMechanical` should be using `ZoneNum`.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here) - *need a case that uses Controller:MechanicalVentilation with a zonelist that does not align with the Zone object and has a DesignSpecification:OutdoorAir that uses ACH*
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by RKStrand**
Pull request overview
---------------------
 - Fixes #9558 
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

There was an improper index used for a calculation in the mechanical ventilation controller that would grab the wrong zone volume.  This has been corrected and a unit test to verify the fix has been created.  A few minor IDD, input file, and doc (IO Ref) fixes from check-ins in a previous release that missed the IDD freeze were tacked onto this work.  No differences in the output as expected because there were no input files in the test suite where this problem was triggered.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/MixedAir.cc [hunk 1]
```diff
@@ -4208,8 +4208,8 @@ void VentilationMechanicalProps::CalcMechVentController(
                     ZoneOAArea =
                         curZone.FloorArea * curZone.Multiplier * curZone.ListMultiplier * this->ZoneOAAreaRate(ZoneIndex) * curZoneOASchValue;
                     ZoneOAFlow = curZone.Multiplier * curZone.ListMultiplier * this->ZoneOAFlowRate(ZoneIndex) * curZoneOASchValue;
-                    ZoneOAACH = curZone.Multiplier * curZone.ListMultiplier *
-                                (this->ZoneOAACHRate(ZoneIndex) * state.dataHeatBal->Zone(ZoneIndex).Volume) * curZoneOASchValue / 3600.0;
+                    ZoneOAACH =
+                        curZone.Multiplier * curZone.ListMultiplier * (this->ZoneOAACHRate(ZoneIndex) * curZone.Volume) * curZoneOASchValue / 3600.0;
```
Comment (Myoldmopar) [inline src/EnergyPlus/MixedAir.cc:4212]: Got it, so yeah, `curZone` is based on `ZoneNum`, `ZoneIndex` should not have been used here.  Works for me.

File: doc/input-output-reference/src/overview/group-surface-construction-elements.tex [hunk 2]
```diff
@@ -4120,7 +4120,7 @@ \subsubsection{Inputs}\label{inputs-38}
...
+This field defines the distance between adjacent hydronic tubes spaced in the direction perpendicular to the main direction of heat transfer. The value for this parameter must be greater than or equal to 0.01m (or a tube spacing of 1 cm) and less than or equal to 1.0m (or a tube spacing of 1m). Note that this parameter is only used for two-dimensional solutions (see previous field) or when the user requests that the tube length for a hydronic radiant system ( \textbf{\hyperref[zonehvaclowtemperatureradiantvariableflow]{variable flow}} or  \textbf{\hyperref[zonehvaclowtemperatureradiantconstantflow]{constant flow}}) be autosized.  In the case of autosizing the tube length, this parameter is used along with the dimensions of the surface to approximate the tube length.
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-surface-construction-elements.tex:4123]: I don't see from the PR description anything about the tube size issue.  Is that just something you identified while fixing the zone index issue and added it here?

File: tst/EnergyPlus/unit/MixedAir.unit.cc [hunk 3]
```diff
@@ -5514,6 +5514,100 @@ TEST_F(EnergyPlusFixture, MechVentController_VRPNoCap)
...
+
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand.allocate(2);                // Necessary for CalcMechVentController
+    state->dataSize->SysSizingRunDone = true;                                    // Indicate that a system sizing run has been performed
+    state->dataHeatBal->Zone(1).Volume = 100.0;                                  // Zone 1 total floor area
+    state->dataHeatBal->Zone(2).Volume = 50.0;                                   // Zone 2 total floor area
+    state->dataSize->FinalSysSizing.allocate(1);                                 // Create instance of system sizing info
+    state->dataMixedAir->VentilationMechanical(1).SysDesOA = ExpectedOAMassFlow; // Set design outdoor air flow rate
+    state->dataSize->CurSysNum = 1;                                              // Only one system in this instance
+    state->dataEnvrn->StdRhoAir = 1;                                             // Standard air density assumed to be 1 kg/m3 (simplification)
+    state->dataMixedAir->VentilationMechanical(1).CalcMechVentController(*state, SysMassFlow, OAMassFlow);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/MixedAir.unit.cc:5606]: Nice.

**Review by Myoldmopar — COMMENTED** [review]
The zone index fix is pretty obvious, I'm just a little thrown by the tube sizing changes in this PR as well.  I don't necessarily have a problem with them, but are they intentional?

**Comment by RKStrand** [comment]
@Myoldmopar sorry, should have explained better. I combined in some IDD items that missed a previous release. They had nothing to do with this particular defect but I was trying to avoid having extra PRs. Is it okay to do this or do I need to create a separate branch to do this.

**Comment by Myoldmopar** [comment]
> Is it okay to do this or do I need to create a separate branch to do this.

Yes, this is perfectly OK, please combine away.  Just make sure the PR description has enough detail so that I know what to expect with the changes being made.  :+1:

File: idd/Energy+.idd.in [hunk 4]
```diff
@@ -90281,10 +90283,10 @@ ElectricLoadCenter:Storage:Simple,
        \minimum 0.0
   N2,  \field Nominal Energetic Efficiency for Charging
        \maximum 1.0
-       \minimum 0.0
+       \minimum 0.001
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:90286]: @RKStrand @mjwitte what do you think about this versus like `\minimum> 0`.  I'm not strongly opinionated either way.

File: testfiles/GeneratorswithPV.idf [hunk 5]
```diff
@@ -182,7 +182,7 @@
     1,                       !- Thermal Source Present After Layer Number
     1,                       !- Temperature Calculation Requested After Layer Number
     1,                       !- Dimensions for the CTF Calculation
-    0.0,                     !- Tube Spacing {m}
+    0.1,                     !- Tube Spacing {m}
```
Comment (Myoldmopar) [inline testfiles/GeneratorswithPV.idf:185]: :+1:

File: idd/Energy+.idd.in (hunk 4 above)
Comment (RKStrand) [reply idd/Energy+.idd.in:90286]: @Myoldmopar Well, the code that now exists uses 0.001 as a hard limit for both charging and discharging.  I guess we could eliminate that logic now with the change to the IDD.  If I remember correctly, my thinking was that it is going to divide by the number and I wanted to make sure we avoided the problem of having huge electric loads (an input file set this parameter to zero which ended up being a very tiny number and a massive electric load.  I guess I'm okay with it like this, but I'll wait to hear if @mjwitte has a strong opinion.

Comment (mjwitte) [reply idd/Energy+.idd.in:90286]: No opinion, really. Anything below 0.1 seems pathetic anyway, so I would hope nobody actually models anything this low.

Comment (Myoldmopar) [reply idd/Energy+.idd.in:90286]: :+1:

**Comment by Myoldmopar** [comment]
Alright, with that resolved, let's merge this in.  Thanks @RKStrand

**Pull request merged**